packages/odyssey-react-mui/src/test-selectors/querySelector.ts

```diff
@@ -11,188 +11,238 @@
-          };`,
-      )
-      .join("")}
-
-    \`${string}\`
-  `) as string;
-
-  if (/^\/*(.+)\/$/.test(interpolatedString)) {
-    return eval(interpolatedString) as RegExp;
-  }
-
-  return interpolatedString;
-};
-
-const getByQuerySelector = ({
-  canvas,
-  method,
-  options,
-  role,
-  text,
-}: {
-  canvas: BoundFunctions<typeof queries>;
-  method: "ByRole" | "ByLabelText" | "ByPlaceholderText" | "ByText";
-  options?: ByRoleOptions;
-  role?: Parameters<GetByRole>[1];
-  text?: Parameters<GetByText>[1];
-}) => {
-  if (method === "ByRole") {
-    return canvas.getByRole(
-      // TODO: These should eventually reference `query` as the function identifier.
-      role!,
-      options,
-    );
-  } else if (method === "ByLabelText") {
-    return canvas.getByLabelText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
-      options,
-    );
-  } else if (method === "ByPlaceholderText") {
-    return canvas.getByPlaceholderText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `label` is required when it's `ByLabelText`.
-      options,
-    );
-  } else if (method === "ByText") {
-    return canvas.getByText(
-      // These should eventually reference `query` as the function identifier.
-      text!, // TODO: Use TypeScript `Infer` to ensure `description` is required when it's `ByLabelText`.
-      options,
-    );
-  }
-
-  return null;
-};
-
-export const querySelector = <TestSelectors extends FeatureTestSelector>({
-  canvas,
-  templateArgs: templateArgsProp,
-  testSelectors,
-}: {
+import { getComputedAccessibleText } from "./getComputedAccessibleText";
+import {
+  getByRoleQuerySelector,
+  getByTextQuerySelector,
+  type QueryMethod,
+} from "./getByQuerySelector";
+import { getControlledElement } from "./linkedHtmlSelectors";
+import { ElementError } from "./sanityChecks";
+
+export type QuerySelectorOptions<
+  LocalFeatureTestSelector extends FeatureTestSelector,
+> = LocalFeatureTestSelector extends TestSelector
+  ? Record<
+      keyof LocalFeatureTestSelector["selector"]["options"],
+      string | RegExp
+    >
+  : Record<string, string>;
+
+export type InnerQuerySelectorProps<
+  LocalFeatureTestSelector extends FeatureTestSelector,
+  LocalQueryMethod extends QueryMethod,
+> = {
   /**
-   * Testing Library canvas. This is usually `screen`, but Storybook uses `within(canvas)`.
+   * Testing Library method used to query elements.
    */
-  canvas: BoundFunctions<typeof queries>;
-  templateArgs?: TestSelectors extends TestSelector
-    ? Record<
-        TestSelectors["selector"]["templateVariableNames"][number],
+  queryMethod?: LocalQueryMethod;
+} & (LocalFeatureTestSelector extends {
+  selector: {
+    role: infer Role;
+  };
+}
+  ? Role extends AriaRole[]
```

Comment:
the formatting on these conditionals is not great. If possible I'd like to see these extracted into named types where the branches in the types are more understandable

Reply:
I agree, Prettier does a number on these types. They looked somewhat easier to read originally.

Can you provide an example of what you mean? I had these separated out before, but I didn't end up using the other types anywhere.